daemon/daemon.go

```diff
@@ -385,16 +382,35 @@ func (d *Daemon) useK8sNodeCIDR(nodeName string) error {
 	if err != nil {
 		return err
 	}
-	ciliumIPv4, err := addressing.NewCiliumIPv4(ip.String())
-	if err != nil {
-		return err
-	}
-	ipv6NodeAddress := nodeaddress.IPv6Address.NodeIP().String()
-	err = nodeaddress.SetNodeAddress(ipv6NodeAddress, ciliumIPv4.NodeIP().String(), "")
-	if err != nil {
-		return err
+
+	// Check if the pod CIDR is either IPv4 or IPv6
+	if ip.To4() != nil {
+		ciliumIPv4, err := addressing.NewCiliumIPv4(ip.String())
+		if err != nil {
+			return err
+		}
+		ipv6NodeAddress := nodeaddress.IPv6Address.NodeIP().String()
```

Comment:
I suggest we wait and rebase this on top of #1020 as this will change heavily and you no longer need to also derive the node address from this.